File: finalThroughPJT_server/src/main/java/com/ssafy/cafe/model/service/UserService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import java.util.Map;

import com.ssafy.cafe.model.dto.Comment;
import com.ssafy.cafe.model.dto.User;


public interface UserService {
    /**
     * 사용자 정보를 DB에 저장한다.
     * 
     * @param user
     */
    public void join(User user);

    /**
     * id, pass에 해당하는 User 정보를 반환한다.
     * 
     * @param id
     * @param pass
     * @return
     * 조회된 User 정보를 반환한다.
     */
    public User login(String id, String pass);
    
    /**
     * id에 해당하는 사용자 정보를 삭제한다.
     * @param id
     */
    public void leave(String id);
    
    /**
     * 해당 아이디가 이미 사용 중인지를 반환한다.
     * @param id
     * @return
     */
    public boolean isUsedId(String id);


    public User getInfo(String id);

    void updateUserMoney(User user);
    
    /**
     * 비밀번호 체크
     * @param id
     * @param pw
     * @return
     */
    public boolean chkPw(String id, String pw);
 
    /**
     *  사용자 Token이 없데이트되면 DB에도 수정한다.
     * @param user
     */
    void updateUserToken(User user);
    
    String selectUserId(String token);
    
    String selectUserToken(String id);
    
    public List<String> selectTokens();
    

}

File: finalCoffeeProject/app/src/main/java/com/ssafy/cafe/service/UserService.kt
package com.ssafy.cafe.service

import android.util.Log
import com.ssafy.cafe.dto.Grade
import com.ssafy.cafe.dto.User
import com.ssafy.cafe.util.RetrofitCallback
import com.ssafy.cafe.util.RetrofitUtil
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit

private const val TAG = "UserService"
class UserService {
    fun login(user: User, callback:RetrofitCallback<User>) {
        RetrofitUtil.userService.login(user).enqueue(object : Callback<User> {
            override fun onResponse(call: Call<User>, response: Response<User>) {
                val res = response.body()
                if(response.code() == 200){
                    if(res != null){
                        callback.onSuccess(response.code() , res)
                    }
                }else{
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<User>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    fun join(user:User, callback: RetrofitCallback<Boolean>) {
        RetrofitUtil.userService.insert(user).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val data = response.body()
                if(response.code() == 200) {
                    if(data == true) {
                        callback.onSuccess(response.code(), data)
                    } else {
                        callback.onFailure(response.code())
                    }
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    fun isUsed(id: String, callback: RetrofitCallback<Boolean>) {
        RetrofitUtil.userService.isUsedId(id).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200) {
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    fun getUsers(id:String, callback: RetrofitCallback<HashMap<String, Any>>){
        RetrofitUtil.userService.getInfo(id).enqueue(object : Callback<HashMap<String, Any>>{
            override fun onResponse(
                call: Call<HashMap<String, Any>>,
                response: Response<HashMap<String, Any>>
            ) {
                val data = response.body()
                if(response.code() == 200){
                    if(data!=null){
                       // Log.d(TAG, "onResponse: ${data!!["order"]}")
                        callback.onSuccess(response.code() , data)
                    }else{
                        callback.onFailure(response.code())
                    }
                }


            }

            override fun onFailure(call: Call<HashMap<String, Any>>, t: Throwable) {
                callback.onError(t)
            }

        })
    }
}